packages/proxy/lib/http/response-middleware.ts

```diff
@@ -381,11 +382,73 @@ const MaybePreventCaching: ResponseMiddleware = function () {
   this.next()
 }
 
+const determineIfNeedsMultiDomainHandling = (ctx) => {
+  const previousAUTRequestUrl = ctx.getPreviousAUTRequestUrl()
+
+  // A cookie needs multi-domain handling if it's an AUT request and
+  // either the request itself is cross-origin or the origins between
+  // requests don't match, since the browser won't set them in that
+  // case and if it's secondary-domain -> primary-domain, we don't
+  // recognize the request as cross-origin
+  return (
+    !!ctx.req.isAUTFrame &&
+    (
+      (previousAUTRequestUrl && !cors.urlOriginsMatch(previousAUTRequestUrl, ctx.req.proxiedUrl))
+      || !reqMatchesOriginPolicy(ctx.req, ctx.getRemoteState())
+    )
+  )
+}
+
+const cookieSameSiteRegex = /SameSite=(\w+)/i
+const cookieSecureRegex = /Secure/i
+const cookieSecureSemicolonRegex = /;\s*Secure/i
+
+const ensureSameSiteNone = ({ cookie, browser, isLocalhost, url }) => {
+  debug('original cookie: %s', cookie)
+
+  if (cookieSameSiteRegex.test(cookie)) {
+    debug('change cookie to SameSite=None')
+    cookie = cookie.replace(cookieSameSiteRegex, 'SameSite=None')
+  } else {
+    debug('add SameSite=None to cookie')
+    cookie += '; SameSite=None'
+  }
+
+  const isFirefox = browser.family === 'firefox'
+
+  // Secure is required in Chrome for SameSite=None cookies to be set, but if
+  // in Firefox and http://localhost, it causes the cookie not to be set,
+  // so we need to ensure there is no Secure in that case
+  if (cookieSecureRegex.test(cookie)) {
+    if (isFirefox && isLocalhost && url.protocol === 'http:') {
+      debug('remove Secure from cookie')
+      cookie = cookie.replace(cookieSecureSemicolonRegex, '')
+    }
+  } else if (!isFirefox || url.protocol === 'https:') {
+    debug('add Secure to cookie')
+    cookie += '; Secure'
+  }
+
+  debug('resulting cookie: %s', cookie)
+
+  return cookie
+}
+
 const CopyCookiesFromIncomingRes: ResponseMiddleware = function () {
   const cookies: string | string[] | undefined = this.incomingRes.headers['set-cookie']
+  const needsMultiDomainHandling = this.config.experimentalMultiDomain && determineIfNeedsMultiDomainHandling(this)
+  const browser = this.getCurrentBrowser() || { family: null }
+  const url = new URL(this.req.proxiedUrl)
+  const isLocalhost = uri.isLocalhost(url)
```

Comment:
these 3 lines should go inside of `if (cookies)`, since they are not used outside of that scope

Reply:
Addressed in https://github.com/cypress-io/cypress/pull/20450/commits/e88519acefe68f5966b516d9a018fdf63ea3b6cf